internal/pgbouncer/certificates.go

```diff
@@ -80,8 +88,46 @@ func frontendCertificate(
 					Path: certFrontendProjectionPath,
 				},
 			},
+		}}
+	}
+
+	var items []corev1.KeyToPath
+	result := custom.DeepCopy()
+
+	for i := range result.Items {
+		// The custom projection expects Path to match typical Keys.
+		switch result.Items[i].Path {
+		case tlsAuthoritySecretKey:
+			result.Items[i].Path = certFrontendAuthorityProjectionPath
+			items = append(items, result.Items[i])
+
+		case tlsCertificateSecretKey:
+			result.Items[i].Path = certFrontendProjectionPath
+			items = append(items, result.Items[i])
+
+		case tlsPrivateKeySecretKey:
+			result.Items[i].Path = certFrontendPrivateKeyProjectionPath
+			items = append(items, result.Items[i])
+		}
+	}
+
```

Comment:
We don't load the Secret here, so we don't know what keys it contains. You're right, listing them explicitly might prevent the Pod from starting. The user can set `spec.proxy.pgBouncer.customTLSSecret.optional` to start the Pod anyway, but ultimately, PgBouncer must read these files to successfully start.

Both before and after this change, PgBouncer doesn't work if the Secret is incomplete.

Similarly, we use whatever key/paths the user has specified. If they don't specify enough (e.g. just one) we create the Pod and PgBouncer fails to start.

These are the downsides to putting an entire `SecretProjection` in our spec. We thought it was acceptable at the time.

A webhook could validate this; do similar digging in the projection as this code does here. I think a shape that makes it more clear what fields are required and optional is better long-term.

Reply:
Completely agree this aligns with current behavior, so no changes specifically needed for this PR.  Was just thinking that since we're now actually parsing the projection the provide and can now actually know ahead of time that the Deployment will definitely fail, maybe there is a potential value-add here in providing the user with an error, warning event, etc.

However, the more I thought of it, my mind kept going (once again!) back to a webhook as the proper solution to prevent this scenario in the first place.  So definitely fine with adding a webhook `TODO` here to track accordingly.

Reply:
Added a comment above this loop with a TODO about validation.